napari/_vispy/vispy_surface_layer.py

```diff
@@ -51,25 +46,50 @@ def _on_slice_data_change(self, event=None):
         self._on_scale_change()
         self._on_translate_change()
 
-    def _on_colormap_change(self, event=None):
-        cmap = self.layer.colormap[1]
-        if self.layer.gamma != 1:
+    def _on_colormap_change(self, colormap):
+        """Receive layer model colormap change event and update visual.
+
+        Parameters
+        ----------
+        colormap : tuple
+            colormap name and colormap
+        """
+
+        name, cmap = ensure_colormap_tuple(colormap)
+        # Once #1842 and #1844 from vispy are released and gamma adjustment is
+        # done on the GPU this can be dropped
+        self._raw_cmap = cmap
+        if self._gamma != 1:
             # when gamma!=1, we instantiate a new colormap with 256 control
             # points from 0-1
-            cmap = Colormap(cmap[np.linspace(0, 1, 256) ** self.layer.gamma])
-        if self.layer.dims.ndisplay == 3:
-            self.node.view_program['texture2D_LUT'] = (
-                cmap.texture_lut() if (hasattr(cmap, 'texture_lut')) else None
-            )
-        self.node.cmap = cmap
+            node_cmap = Colormap(cmap[np.linspace(0, 1, 256) ** self._gamma])
+        else:
+            node_cmap = cmap
+        self.node.cmap = node_cmap
+
+    def _on_contrast_limits_change(self, contrast_limits):
+        """Receive layer model contrast limits change event and update visual.
+
+        Parameters
+        ----------
+        contrast_limits : tuple
+            Contrast limits.
+        """
+
+        self.node.clim = contrast_limits
+
+    def _on_gamma_change(self, gamma):
+        """Receive layer model gamma change event and update visual.
 
-    def _on_contrast_limits_change(self, event=None):
-        self.node.clim = self.layer.contrast_limits
+        Parameters
+        ----------
+        gamma : float
+            New gamma value.
+        """
 
-    def _on_gamma_change(self, event=None):
-        self._on_colormap_change()
+        self._on_colormap_change(self.layer.colormap)
```

Comment:
This code block needs to be duplicated too https://github.com/napari/napari/blob/master/napari/_vispy/vispy_image_layer.py#L186-L195